ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx

```diff
@@ -1223,4 +1429,1134 @@ describe(RuntimeConfigurationPanel.name, () => {
+
+    clickExpectedButton('Next');
+
+    expect(
+      screen.getByText(
+        /your environment currently has a reattachable disk, which will be unused after you apply this update\./i
+      )
+    ).toBeTruthy();
+
+    // Default option should be NOT to delete.
+    clickExpectedButton('Next');
+    clickExpectedButton('Update');
+    runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
+
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual(RuntimeStatus.CREATING);
+    });
+    expect(disksApiStub.disk?.name).toEqual(disk.name);
+  });
+
+  it('should prevent runtime creation when running cost is too high for initial credits', async () => {
+    setCurrentRuntime(null);
+    const { container } = component();
+    clickExpectedButton('Customize');
+    const createButton = screen.getByRole('button', { name: 'Create' });
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(150);
+    // This should make the cost about $50 per hour.
+    await pickNumWorkers(200);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(2);
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should prevent runtime creation when worker count is invalid', async () => {
+    setCurrentRuntime(null);
+    const { container } = component();
+    clickExpectedButton('Customize');
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(DATAPROC_MIN_DISK_SIZE_GB);
+    const createButton = screen.getByRole('button', { name: 'Create' });
+    await pickNumWorkers(0);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(1);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(2);
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should prevent runtime creation when running cost is too high for paid tier', async () => {
+    setCurrentRuntime(null);
+    currentWorkspaceStore.next({
+      ...workspaceStubs[0],
+      accessLevel: WorkspaceAccessLevel.WRITER,
+      cdrVersionId: CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION_ID,
+    });
+    const { container } = component();
+
+    clickExpectedButton('Customize');
+
+    await pickComputeType(container, ComputeType.Dataproc);
+
+    await pickStandardDiskSize(DATAPROC_MIN_DISK_SIZE_GB);
+    // This should make the cost about $140 per hour.
+    await pickNumWorkers(600);
+    let createButton = screen.getByRole('button', { name: 'Create' });
+    expectButtonElementEnabled(createButton);
+
+    // This should make the cost around $160 per hour.
+    await pickNumWorkers(700);
+    createButton = screen.getByRole('button', { name: 'Create' });
+    // We don't want to disable for user provided billing. Just put a warning.
+    expectButtonElementEnabled(createButton);
+    expect(
+      screen.findByText(
+        'Your runtime is expensive. Are you sure you wish to proceed?'
+      )
+    ).toBeTruthy();
```

Comment:
nit: I'd prefer `.toBeInTheDocument()` / `.not.toBeInTheDocument()` for cases like this